src/fs_io.cc

```diff
@@ -138,21 +138,11 @@ diskCombineWrites(_fde_disk *fdd)
         for (dwrite_q *q = fdd->write_q; q != nullptr; q = q->next)
             len += q->len - q->buf_offset;
 
-        dwrite_q *wq = (dwrite_q *)memAllocate(MEM_DWRITE_Q);
-
+        auto *wq = new dwrite_q;
         wq->buf = (char *)xmalloc(len);
+        wq->free_func = cxx_xfree; // dwrite_q buffer xfree()
 
-        wq->len = 0;
-
-        wq->buf_offset = 0;
-
-        wq->next = nullptr;
-
-        wq->free_func = cxx_xfree;
-
-        while (fdd->write_q != nullptr) {
-            dwrite_q *q = fdd->write_q;
-
+        while (auto *q = fdd->write_q) {
```

Comment:
Here and in all similar changed contexts, please improve const-correctness:

```suggestion
        while (const auto q = fdd->write_q) {
```

Reply:
done in [0cd7a7c](https://github.com/squid-cache/squid/pull/1873/commits/0cd7a7c556e5b55989982cc7160a0016e3d89993)